esy-lib/Curl.ml

```diff
@@ -14,18 +14,14 @@ type url = string
 
 let parseStdout stdout =
   let open Run.Syntax in
-  match String.cut ~rev:true ~sep:"\n" stdout with
-  | Some (stdout, meta) ->
-    let%bind meta = Json.parseStringWith Meta.of_yojson meta in
+  let%bind meta = Json.parseStringWith Meta.of_yojson stdout in
     return (stdout, meta)
-  | None ->
-    error "unable to parse metadata from a curl response"
 
 let runCurl cmd =
   let cmd = Cmd.(
     cmd
     % "--write-out"
-    % {|\n{"code": %{http_code}}|}
```

Comment:
This `\n` was causing problems with Windows - it would actually emit the `\n` string as opposed to a newline. I don't think this will be reliable cross-platform, as the new line will end up being different.

I removed this and the `String.cut` above, and it seemed to work as expected - but I could be missing a case. Was there a case that you specifically added this for that I should test, @andreypopp ?

Reply:
The `\n` is needed in case there's no response body iirc. I don't have a test for it though.

Why `\n` is emitted as a literal string and not a newline? Is this because of some quoting in bash?

Reply:
Also from `man curl`:
```
              NOTE:  The  %-symbol  is  a special symbol in the win32-environment, where all occur-
              rences of % must be doubled when using this option.
```

Reply:
Maybe it would be cleaner always to use `--output FILENAME` to output response body to file and use stdout only to get metadata (http response code, ...).

Reply:
Interesting, nice find. I wonder if we get around that `%` issue running in our cygwin-bash shell.

Reply:
I do see a bug in the way we are handling `\` in `esy-bash` - so I'll fix that in bryphe/esy-bash#10 and test it out